### ISSUE TITLE
Improve restriction rules

### ISSUE THREAD

**Issue by lightdiscord** [Popcorn-moe/migi#3]
Currently we have a decorator to check if a user has permission.

But it would be nice to have more restrictions on commands:
* Check user group
* Check channel's name
* Check guild's name
* ...

In decorators, it could be something like
```js
@command(/^command$/i, {
    name: 'command',
    desc: 'My command',
    restrictions: [
        ({ guild }) => guild.name == "Popcorn",
        ({ member }) => member.roles.some(role => role.name == "Admin"),
        ({ channel }) => channel.name == "secret",
        // May work with async functions
        async () => await somethingAsync()
    ]
})
```

(`restrictions` is an array of function that returns true or false)

**Comment by DeltaEvo** [comment]
I don't think the restriction should go inside the `command` decorator
I think we should create a new decorator `restrict`
```js
@restrict(({ guild }) => guild.name == "Popcorn")
@restrict(({ member }) => member.roles.some(({ name }) => name == "Admin"))
...
```

**Comment by lightdiscord** [comment]
Definitely, it will also allow the use of restrictions on listeners :smiley:

**Comment by DeltaEvo** [comment]
Closed by #7
